### PR TITLE
walllet login api added

### PR DESCRIPTION
**Pull request by abiskar-sapkota**

### CODE REVIEW

File: modules/user/user.controllers.js [hunk 1]
```diff
@@ -23,6 +28,29 @@ const mixins = {
...
+        wallet_address: address,
+      },
+    });
+    let data = await this.loginSuccess(user.id, clientIpAddress);
+    if (data.user.roles?.includes('donor')) {
+      let keys = getPrivateKeys('donor');
+
+      data.privateKey = keys.privateKey;
+    } else {
+      let keys = getPrivateKeys('admin');
```
Comment (myanzik) [inline modules/user/user.controllers.js]: No need to send private key if login from wallet. user use their metamask to do transaction.

File: modules/user/user.controllers.js [hunk 2]
```diff
@@ -23,6 +28,29 @@ const mixins = {
...
+    const { address } = WalletUtils.validateSignature(signature, signPayload, {
+      ip: clientIpAddress,
+      secret,
+    });
+    let user = await this.table.findOne({
+      where: {
+        wallet_address: address,
+      },
+    });
+    let data = await this.loginSuccess(user.id, clientIpAddress);
```
Comment (myanzik) [inline modules/user/user.controllers.js]: what does loginSuccess() do?

**Review by myanzik — CHANGES_REQUESTED** [review]
please go through my comments

File: modules/user/user.controllers.js (hunk 2 above)
Comment (abiskar-sapkota) [reply modules/user/user.controllers.js]: login success is called to get access-token after verifying signature

Comment (myanzik) [reply modules/user/user.controllers.js]: in that case we need to throw error if user doesn't exist. right? @abiskar-sapkota

Comment (myanzik) [reply modules/user/user.controllers.js]: @abiskar-sapkota

**Review by myanzik — APPROVED** [review]

**Pull request merged**